### PR TITLE
Todo list. Main app

### PR DESCRIPTION
**Pull request by EmmanuelWLX**
### Summary
- First version of the app
- Basic interaction listed in the card
- What an informative summary!

### Screenshots
todo1
todo2
todo3

### Trello card
https://trello.com/c/zoKOU7MH/1-componentes-i-yet-another-todo-list

### CODE REVIEW

File: React-Native/TodoList/src/app/screens/Main/components/Footer/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react';
+import { View, Text } from 'react-native';
+import styles from './styles';
+
+export default class Footer extends Component {
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Footer/index.js]: make the export at the end

File: React-Native/TodoList/src/app/screens/Main/components/Footer/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  render() {
+    return (
+      <View style={styles.container}>
+        <Text style={styles.removeLabel} onPress={() => this.props.onRemoveCompleted()}>Clean completed items</Text>
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Footer/index.js]: extract onPress function outside the render

File: React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: 'lightblue',
+    padding: 15
+  },
+  removeLabel: {
+    textAlign: 'center',
+    color: 'red'
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js]: add a constant file for colors, called colors.js.. there you can define this one and all you have

File: React-Native/TodoList/src/app/screens/Main/components/Header/styles.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: 'skyblue',
+    padding: 15,
+  },
+  title: {
+    textAlign: 'center',
+    color: 'white',
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Header/styles.js]: same

File: React-Native/TodoList/src/app/screens/Main/components/Header/styles.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: 'skyblue',
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Header/styles.js]: same

File: React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: 'lightblue',
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js]: same

File: React-Native/TodoList/src/app/screens/Main/components/Input/index.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { TextInput } from 'react-native';
+import styles from './styles';
+
+export default class Input extends Component {
+
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Input/index.js]: delete empty line

File: React-Native/TodoList/src/app/screens/Main/components/Input/index.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { TextInput } from 'react-native';
+import styles from './styles';
+
+export default class Input extends Component {
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Input/index.js]: same as before.. export at the end 😉

File: React-Native/TodoList/src/app/screens/Main/components/Input/index.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+  state = {
+    text: ''
+  }
+
+  onChangeText = text => this.setState({ text });
+
+  onSubmitEditing = () => {
+    const { text } = this.state;
+
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Input/index.js]: delete empty line

File: React-Native/TodoList/src/app/screens/Main/components/Input/index.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    if (!text) return;
+
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Input/index.js]: delete empty line

File: React-Native/TodoList/src/app/screens/Main/components/Input/styles.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  input: {
+    padding: 15,
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Input/styles.js]: you are already using this value `15` in another file.. so you could make a constant file like `dimensions.js` for all repeated constans for size

File: React-Native/TodoList/src/app/screens/Main/components/List/index.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import { ScrollView, Text, View, CheckBox } from 'react-native';
+import styles from './styles';
+
+export default class List extends Component {
+
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/List/index.js]: delete empty line

File: React-Native/TodoList/src/app/screens/Main/components/List/index.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import { ScrollView, Text, View, CheckBox } from 'react-native';
+import styles from './styles';
+
+export default class List extends Component {
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/List/index.js]: make the export at the end

File: React-Native/TodoList/src/app/screens/Main/components/List/styles.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  container: {
+    flex: 1
+  },
+  horizontalContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between'
+  },
+  removeButton: {
+    color: 'red',
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/List/styles.js]: same, extract color

File: React-Native/TodoList/src/app/screens/Main/components/List/styles.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    fontSize: 28
+  },
+  buttonsContainer: {
+    alignItems: 'center',
+    borderLeftWidth: 1,
+    borderLeftColor: 'lightblue',
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/List/styles.js]: same

File: React-Native/TodoList/src/app/screens/Main/index.js [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+import { View } from 'react-native';
+import Header from './components/Header';
+import Input from './components/Input';
+import List from './components/List';
+import Footer from './components/Footer';
+import todoActions from '../../../redux/Todo/actions';
+import styles from './styles';
+
+class Main extends Component {
+
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/index.js]: delete empty line

File: React-Native/TodoList/src/app/screens/Main/index.js [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+  // Chequear esto, todavía no hice nada para que el input me llene ese item que pido
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/index.js]: delete comment.. remember to make it in english

File: React-Native/TodoList/src/app/screens/Main/index.js [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+  addItem = item => {
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/index.js]: i don't think the point in defining this functions... directly pass the functions in props..

File: React-Native/TodoList/src/app/screens/Main/styles.js [hunk 19]
```diff
@@ -0,0 +1,13 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  divider: {
+    height: 1,
+    backgroundColor: 'whitesmoke',
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/styles.js]: extract color

File: React-Native/TodoList/src/redux/store.js [hunk 20]
```diff
@@ -0,0 +1,25 @@
+import { createStore, compose, combineReducers, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+// import Reactotron from 'reactotron-react-native';
```
Comment (julibel) [inline React-Native/TodoList/src/redux/store.js]: delete comment

File: React-Native/TodoList/src/redux/Todo/reducer.js [hunk 21]
```diff
@@ -0,0 +1,38 @@
+import immutable from 'seamless-immutable';
+import shortid from 'shortid';
```
Comment (julibel) [inline React-Native/TodoList/src/redux/Todo/reducer.js]: use uppercase: `shortId`

File: React-Native/TodoList/src/redux/store.js [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+
+middlewares.push(thunk);
+
+enhancers.push(applyMiddleware(...middlewares));
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // eslint-disable-line no-underscore-dangle
+
+export default createStore(reducers, composeEnhancers(...enhancers));
+
+// const createAppropriateStore = __DEV__ ? Reactotron.createStore : createStore;
```
Comment (julibel) [inline React-Native/TodoList/src/redux/store.js]: delete comments if you are not using it

File: React-Native/TodoList/src/redux/store.js [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+// const store = createAppropriateStore(reducers, compose(...enhancers));
+
+// export default store;
```
Comment (julibel) [inline React-Native/TodoList/src/redux/store.js]: same

File: React-Native/TodoList/src/redux/store.js [hunk 24]
```diff
@@ -0,0 +1,25 @@
...
+middlewares.push(thunk);
+
+enhancers.push(applyMiddleware(...middlewares));
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // eslint-disable-line no-underscore-dangle
+
+export default createStore(reducers, composeEnhancers(...enhancers));
+
+// const createAppropriateStore = __DEV__ ? Reactotron.createStore : createStore;
+// const store = createAppropriateStore(reducers, compose(...enhancers));
```
Comment (julibel) [inline React-Native/TodoList/src/redux/store.js]: same

File: React-Native/TodoList/src/redux/store.js (hunk 23 above)
Comment (julibel) [reply React-Native/TodoList/src/redux/store.js]: are you sure you don't need this?

**Review by julibel — CHANGES_REQUESTED** [review]

File: React-Native/TodoList/src/redux/store.js (hunk 23 above)
Comment (EmmanuelWLX) [reply React-Native/TodoList/src/redux/store.js]: i couldn't make reactotron work :(

File: React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js [hunk 25]
```diff
@@ -1,13 +1,15 @@
 import { StyleSheet } from 'react-native';
+import { LIGHTBLUE, RED } from '../../../../../constants/colors';
```
Comment (julibel) [inline React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js:2]: is ok.. as convention we have colors in lower case.. But it's not a problem, lets leave it like this here

**Review by julibel — APPROVED** [review]

File: React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js (hunk 25 above)
Comment (EmmanuelWLX) [reply React-Native/TodoList/src/app/screens/Main/components/Footer/styles.js:2]: oh... i get it

**Pull request merged**